### PR TITLE
Fixes estimated Pending Rewards value decreased unexpectedly

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/16678

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See https://github.com/brave/brave-browser/issues/16678

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/account/ad_rewards/ad_rewards.cc [hunk 1]
```diff
@@ -289,6 +289,9 @@ void AdRewards::OnGetPayments(const UrlResponse& url_response) {
     return;
   }
 
+  unreconciled_estimated_pending_rewards_ = 0.0;
+  ConfirmationsState::Get()->Save();
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/account/ad_rewards/ad_rewards.cc:293]: Set `unreconciled_estimated_pending_rewards_` to `0.0` if we successfully persist the payment balances. As the issue was that `unreconciled_estimated_pending_rewards_` was only reset to `0.0` if both `GetPayment` and `GetAdGrant` endpoints were successful

File: vendor/bat-native-ads/src/bat/ads/internal/account/confirmations/confirmations_state.cc [hunk 2]
```diff
@@ -676,6 +676,14 @@ bool ConfirmationsState::ParseAdRewardsFromDictionary(
     return false;
   }
 
+  const base::Value* unblinded_tokens =
+      dictionary->FindListKey("unblinded_payment_tokens");
+  if (unblinded_tokens && unblinded_tokens->GetList().empty()) {
+    // Migration path for https://github.com/brave/brave-browser/issues/16678
+    ad_rewards_dictionary->SetDoubleKey(
+        "unreconciled_estimated_pending_rewards", 0.0);
+  }
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/account/confirmations/confirmations_state.cc:685]: Migration path to set `unreconciled_estimated_pending_rewards` to `0.0` if the user has no unreconciled unblinded payment tokens

File: vendor/bat-native-ads/data/test/confirmations_issue_16678.json [hunk 3]
```diff
@@ -0,0 +1,139 @@
+{
```
Comment (tmancey) [inline vendor/bat-native-ads/data/test/confirmations_issue_16678.json:1]: `confirmations.json` to replicate @LaurenWags state

**Review by moritzhaller — APPROVED** [review]

File: vendor/bat-native-ads/src/bat/ads/internal/account/ad_rewards/ad_rewards_test.cc [hunk 4]
```diff
@@ -25,7 +25,7 @@ class BatAdsAdRewardsIntegrationTest : public UnitTestBase {
 };
 
 TEST_F(BatAdsAdRewardsIntegrationTest,
-       GetAdRewardsNotUpdatedDueToLossOfPrecisionIssue15801) {
+       FailedToUpdateAdRewardsDueToLossOfPrecisionIssue15801) {
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/account/ad_rewards/ad_rewards_test.cc:28]: Renamed for consistency with `FailedToUpdateAdRewardsDueToNotZeroingUnreconciledEstimatedPendingRewardsIssue16678`

**Review by evq — APPROVED** [review]
lgtm

**Comment by LaurenWags** [comment]
Verified using

```
Brave	1.28.46 Chromium: 92.0.4515.70 (Official Build) nightly (x86_64)
Revision	2df01a82398e62190b2aae01aaad7ca3d47aba01-refs/branch-heads/4515@{#923}
OS	macOS Version 10.15.7 (Build 19H1217)
```

Verified test plan from https://github.com/brave/brave-core/pull/9286#issue-681014696.

Confirmed "Current earnings this month (estimated)" shows as expected for June 30th.
Confirmed "Current earnings this month (estimated)" shows as expected for July 1st (before viewing any ads).
Confirmed "BAT arriving in 5 days" message showed expected value of 1.8375 which matches the "Current earnings this month (estimated)" from June 30th.

June 30th | July 1st
----- | -----
 | 

Confirmed able to view an ad and the "Current earnings this month (estimated)" increments as expected.
Confirmed the Ads received this month also increments as expected.
Confirmed that restarting the browser after viewing an ad retained the expected information on brave://rewards ads panel.



Note, unable to confirm NTP widget shows expected information due to C92 issue with NTP not displaying properly. (Logged separately)

Additionally, while running through this test on my personal profile, I encountered and logged https://github.com/brave/brave-browser/issues/16744.

**Pull request merged**